src/WorkerChannel.ts

```diff
@@ -227,11 +227,16 @@ export class WorkerChannel implements IWorkerChannel {
       // explicitly set outputData to empty array to concat later
       response.outputData = [];
 
+      // As legacy behavior, falsy values get serialized to `null` in AzFunctions.
+      // This breaks Durable Functions expectations so we check if we're serializing
+      // for durable and, if so, ensure falsy values get serialized.
+      let isDurableBinding = info?.bindings?.name?.type == 'activityTrigger';
+
       try {
-        if (result) {
+        if (result || (isDurableBinding && result != null)) {

```

Comment:
if `isDurableBinding == true` the `result` can be `undefined` or `''`?

Reply:
That's correct, in durable bindings such as `activityTrigger`s, the result can be 0, false, and "". This is because Durable Function activities are meant to be abstractions similar to regular function invocations, so the programming model should, in theory, allow for any output value that would be valid to return in regular JavaScript _and_ is JSON serializable. This includes the values: 0, false, and `""`. This was also a behavior we used to have in the past, **but it stopped working recently**, as described here: https://github.com/Azure/azure-functions-durable-js/issues/50

I've also expanded my comments in the PR to make this clearer :) Please let me know if there's anything else I can clarify!

As for serializing `undefined`, I think that gets serialized as `null` anyways. I can certainly write code to explicitly avoid serializing `undefined` though. Please let me know if you would prefer I added that.